packages/compiler/core/checker.ts

```diff
@@ -1748,18 +1747,33 @@ export function createChecker(program: Program): Checker {
     if (!sym) return sym;
 
     if (sym.flags & SymbolFlags.DuplicateUsing) {
-      reportAmbiguousIdentifier(node, [...((table.duplicates.get(sym) as any) ?? [])]);
+      const duplicates: { sym: Sym; isGlobal: boolean }[] = [];
+      for (const symbol of [...(table.duplicates.get(sym) ?? [])]) {
+        duplicates.push({ sym: symbol, isGlobal: false });
+      }
+      reportAmbiguousIdentifier(node, duplicates);
       return sym;
     }
     return getMergedSymbol(sym);
   }
 
-  function reportAmbiguousIdentifier(node: IdentifierNode, symbols: Sym[]) {
-    const duplicateNames = symbols.map(getFullyQualifiedSymbolName).join(", ");
+  function reportAmbiguousIdentifier(
+    node: IdentifierNode,
+    symbols: { sym: Sym; isGlobal: boolean }[]
+  ) {
+    let duplicateNames: string[] = [];
+    for (const data of symbols) {
+      const name = getFullyQualifiedSymbolName(data.sym);
```

Comment:
Doesn't checking `data.parent === undefined` would be equivalent to knowing its in the globalScope? and then you don't need to explicitly pass that to `reportAmbiguousIdentifier`

Reply:
ah also need to check `sym.symbolSource`, as Nick says in the other thread, `getFullyQualifiedSymbolName` basically builds the full path so it should be able to know if its at the root or not with the information it gets already